src/platform_impl/windows/event_loop.rs

```diff
@@ -547,115 +580,108 @@ lazy_static! {
     static ref SEND_WAIT_THREAD_ID_MSG_ID: u32 = {
         unsafe {
-            winuser::RegisterWindowMessageA("Winit::SendWaitThreadId\0".as_ptr() as *const i8)
+            RegisterWindowMessageA("Winit::SendWaitThreadId\0".as_ptr() as PSTR)
         }
     };
     /// lparam points to a `Box<Instant>` signifying the time `PROCESS_NEW_EVENTS_MSG_ID` should
     /// be sent.
     static ref WAIT_UNTIL_MSG_ID: u32 = {
         unsafe {
-            winuser::RegisterWindowMessageA("Winit::WaitUntil\0".as_ptr() as *const i8)
+            RegisterWindowMessageA("Winit::WaitUntil\0".as_ptr() as PSTR)
         }
     };
     static ref CANCEL_WAIT_UNTIL_MSG_ID: u32 = {
         unsafe {
-            winuser::RegisterWindowMessageA("Winit::CancelWaitUntil\0".as_ptr() as *const i8)
+            RegisterWindowMessageA("Winit::CancelWaitUntil\0".as_ptr() as PSTR)
         }
     };
     // Message sent by a `Window` when it wants to be destroyed by the main thread.
     // WPARAM and LPARAM are unused.
     pub static ref DESTROY_MSG_ID: u32 = {
         unsafe {
-            winuser::RegisterWindowMessageA("Winit::DestroyMsg\0".as_ptr() as LPCSTR)
+            RegisterWindowMessageA("Winit::DestroyMsg\0".as_ptr() as PSTR)
         }
     };
     // WPARAM is a bool specifying the `WindowFlags::MARKER_RETAIN_STATE_ON_SIZE` flag. See the
     // documentation in the `window_state` module for more information.
     pub static ref SET_RETAIN_STATE_ON_SIZE_MSG_ID: u32 = unsafe {
-        winuser::RegisterWindowMessageA("Winit::SetRetainMaximized\0".as_ptr() as LPCSTR)
-    };
-    static ref THREAD_EVENT_TARGET_WINDOW_CLASS: Vec<u16> =  {
-        use std::ffi::OsStr;
-        use std::os::windows::ffi::OsStrExt;
-
-        OsStr::new("Winit Thread Event Target")
-            .encode_wide()
-            .chain(Some(0).into_iter())
-            .collect()
+        RegisterWindowMessageA("Winit::SetRetainMaximized\0".as_ptr() as PSTR)
     };
+    static ref THREAD_EVENT_TARGET_WINDOW_CLASS: Vec<u16> = util::encode_wide("Winit Thread Event Target");
 }
 
 fn create_event_target_window<T: 'static>() -> HWND {
     unsafe {
-        let class = winuser::WNDCLASSEXW {
-            cbSize: mem::size_of::<winuser::WNDCLASSEXW>() as UINT,
+        let class = WNDCLASSEXW {
+            cbSize: mem::size_of::<WNDCLASSEXW>() as u32,
             style: 0,
             lpfnWndProc: Some(thread_event_target_callback::<T>),
             cbClsExtra: 0,
             cbWndExtra: 0,
-            hInstance: libloaderapi::GetModuleHandleW(ptr::null()),
-            hIcon: ptr::null_mut(),
-            hCursor: ptr::null_mut(), // must be null in order for cursor state to work properly
-            hbrBackground: ptr::null_mut(),
-            lpszMenuName: ptr::null(),
-            lpszClassName: THREAD_EVENT_TARGET_WINDOW_CLASS.as_ptr(),
-            hIconSm: ptr::null_mut(),
+            hInstance: GetModuleHandleW(std::ptr::null_mut()),
```

Comment:
Sounds like `winit` could benefit from enabling the default-allowed [`unused-qualifications`](https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#unused-qualifications) lint so that you don't have to point these all out by hand - could also enable it for just this file for the time being though.